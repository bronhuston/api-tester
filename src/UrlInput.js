import React from 'react';
import PropTypes from 'prop-types';

class UrlInput extends React.Component {
  render() {
    const { url } = this.props;
    return <label>
      URL
      <input type="text" value={url} onChange={e => this.props.onChange(e)} />
    </label>
  }
}

UrlInput.propTypes = {
  url: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default UrlInput;