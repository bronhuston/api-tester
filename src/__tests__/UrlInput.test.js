import React from 'react';
import ReactDOM from 'react-dom';
import UrlInput from '../UrlInput';
import { shallow } from 'enzyme';


it('renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UrlInput url={''} onChange={() => {}} />, div);
});

it('uses url to populate input', () => {
  const url = 'http://test.com';
  const wrapper = shallow(<UrlInput url={url} onChange={() => {}} />);
  expect(wrapper).toMatchSnapshot();

  wrapper.find('input[type="text"]').simulate('change', { target: { value: 'http' } });
  expect(wrapper.state()).toEqual({ url: 'http' });
});