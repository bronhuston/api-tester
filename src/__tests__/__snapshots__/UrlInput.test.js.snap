// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses url to populate input 1`] = `
<label>
  URL
  <input
    onChange={[Function]}
    type="text"
    value="http://test.com"
  />
</label>
`;
